tests/integration/security/rbac/rbac_v2_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/apps"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/util/connection"
+	"istio.io/istio/pkg/test/util/policy"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/util"
+)
+
+const (
+	rbacClusterConfigTmpl = "testdata/istio-clusterrbacconfig.yaml.tmpl"
+	rbacV2RulesTmpl       = "testdata/istio-rbac-v2-rules.yaml.tmpl"
+)
+
+var (
+	inst                 istio.Instance
+	successIfAuthEnabled bool
+)
+
+func TestMain(m *testing.M) {
+	framework.Main("rbac_v2_test", m, istio.SetupOnKube(&inst, setupConfig))
+}
+
+func setupConfig(cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+	//cfg.Values["global.mtls.enabled"] = "true"
+	if cfg.IsMtlsEnabled() {
+		successIfAuthEnabled = true
+	} else {
+		successIfAuthEnabled = false
+	}
+	cfg.Values["sidecarInjectorWebhook.rewriteAppHTTPProbe"] = "true"
+}
+
+func TestRBACV2(t *testing.T) {
+	ctx := framework.NewContext(t)
+	defer ctx.Done(t)
+	ctx.RequireOrSkip(t, environment.Kube)
+
+	env := ctx.Environment().(*kube.Environment)
+	g := galley.NewOrFail(t, ctx, galley.Config{})
+	p := pilot.NewOrFail(t, ctx, pilot.Config{
+		Galley: g,
+	})
+	appInst := apps.NewOrFail(t, ctx, apps.Config{Pilot: p, Galley: g})
+
+	appA, _ := appInst.GetAppOrFail("a", t).(apps.KubeApp)
+	appB, _ := appInst.GetAppOrFail("b", t).(apps.KubeApp)
+	appC, _ := appInst.GetAppOrFail("c", t).(apps.KubeApp)
+	appD, _ := appInst.GetAppOrFail("d", t).(apps.KubeApp)
+
+	connections := []connection.Connection{
+		// Port 80 is where HTTP is served, 90 is where TCP is served.
+		// In RBAC, ExpectedSuccess = true means that the client receives a valid, non-RBAC denied response from the server.
+		{To: appA, From: appB, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/xyz", ExpectedSuccess: false},
```

Comment:
Can we check it's actually denied by RBAC filter when it's denied? The RBAC v1 test will check the response code is "403". This is important so that we know for sure it's denied by our RBAC filter, not by some random failures.

Reply:
Makes sense. Will do.